library/common/extensions/key_value/platform/config.h

```diff
@@ -26,21 +26,21 @@ class PlatformInterface {
 class PlatformKeyValueStore : public KeyValueStoreBase {
 public:
   PlatformKeyValueStore(Event::Dispatcher& dispatcher, std::chrono::milliseconds save_interval,
-                        PlatformInterface& platform_interface, uint64_t max_entries,
-                        const std::string& key);
+                        std::unique_ptr<PlatformInterface>&& platform_interface,
+                        uint64_t max_entries, const std::string& key);
   // KeyValueStore
   void flush() override;
 
 private:
-  PlatformInterface& platform_interface_;
+  // TODO(alyssawilk, goaway) the default PlatformInterface should do up calls through Java and this
+  // can be moved to a non-optional reference.
+  std::unique_ptr<PlatformInterface> platform_interface_;
```

Comment:
If we're making this optional, I think we want to error handle it being null (ENVOY_BUG a warning), WDYT?

Reply:
Changed to a non-optional static singleton.